repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -198,13 +203,30 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                     }
 
                     for (String asset : assets) {
-                        if (asset.contains("*") || asset.contains("?")) {
-                            //DG-898 Bug fix
+                        /*
+                        * We are introducing a hierarchical filter for qualifiedName.
+                        * This requires a migration of existing data to have a hierarchical qualifiedName.
+                        * So this will only work if migration is done, upon migration completion we will set the feature flag to true
+                        * This will be dictated by the feature flag ENABLE_PERSONA_HIERARCHY_FILTER
+                        */
+
+                        // If asset resource ends with /* then add it in hierarchical filter
+                        boolean isHierarchical = asset.endsWith("/*");
+                        if (isHierarchical) {
+                            asset = asset.substring(0, asset.length() - 2);
+                        }
+                        boolean isWildcard = asset.contains("*") || asset.contains("?");
+                        if (isWildcard) {
                             allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset)));
+                        } else if (useHierarchicalQualifiedNameFilter) {
+                            metadataPolicyQualifiedNames.add(asset);
                         } else {
                             terms.add(asset);
                         }
-                        allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset + "/*")));
+
+                        if (!useHierarchicalQualifiedNameFilter || isWildcard) {
+                            allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset + "/*")));
+                        }
```

Comment:
Have you ignored `getPolicyActions(policy).contains(ACCESS_READ_PERSONA_DOMAIN)` case intentionally or it is a miss ?
For read access in Persona Domain policies, we pass `ACCESS_READ_PERSONA_DOMAIN` permission

Reply:
Actually only considered this logic for persona metadata policies for now actually, added logic for domain policies in case we use it in future. Right now metadata policies are the priority.

So we intentionally ignored domain policies for now,